File: CfxGenerator/CustomSignatures/SignatureWithStructArray.vb
Public Class SignatureWithStructArray
    Inherits Signature

    Private arrayIndex As Integer
    Private countIndex As Integer

    Private m_publicArguments As Argument()

    Public Sub New(parent As ISignatureParent, sd As Parser.SignatureData, api As ApiTypeBuilder, arrayIndex As Integer, countIndex As Integer)
        MyBase.New(parent, sd, api)
        Me.arrayIndex = arrayIndex
        Me.countIndex = countIndex

        Dim list = New List(Of Argument)
        For i = 0 To Arguments.Length - 1
            If i <> arrayIndex AndAlso i <> countIndex Then
                list.Add(Arguments(i))
            ElseIf i = arrayIndex Then
                Dim a = New Argument(Arguments(arrayIndex), New CefStructArrayType(Arguments(arrayIndex), Arguments(countIndex)))
                list.Add(a)
                Arguments(i) = a
            End If
        Next
        m_publicArguments = list.ToArray()
    End Sub

    Public Overrides ReadOnly Property ManagedArguments As Argument()
        Get
            Return m_publicArguments
        End Get
    End Property


    Public Overrides Sub DebugPrintUnhandledArrayArguments()
    End Sub

End Class

File: CfxGenerator/CustomSignatures/SignatureWithStructArray.cs
// Copyright (c) 2014-2017 Wolfgang Borgsmüller
// All rights reserved.
// 
// This software may be modified and distributed under the terms
// of the BSD license. See the License.txt file for details.

using System.Collections.Generic;

public class SignatureWithStructArray : Signature {

    private Argument[] m_publicArguments;

    public SignatureWithStructArray(Signature s, int arrayIndex, int countIndex)
        : base(s) {

        var list = new List<Argument>();
        for(var i = 0; i <= Arguments.Length - 1; i++) {
            if(i != arrayIndex && i != countIndex) {
                list.Add(Arguments[i]);
            } else if(i == arrayIndex) {
                var a = new Argument(Arguments[arrayIndex], new CefStructArrayType(Arguments[arrayIndex], Arguments[countIndex]));
                list.Add(a);
                Arguments[i] = a;
            }
        }
        m_publicArguments = list.ToArray();
    }

    public override Argument[] ManagedArguments {
        get { return m_publicArguments; }
    }

    public override void DebugPrintUnhandledArrayArguments(string cefName, CefConfigNode cefConfig, CfxCallMode callMode) {
    }
}